### PR TITLE
Solc5.0

### PR DESCRIPTION
**Pull request by MrChico**

### CODE REVIEW

**Review by asymmetric — COMMENTED** [review]
Can we remove the trailing `_` from the name at this point?

File: src/weth9.sol [hunk 1]
```diff
@@ -28,27 +28,27 @@ contract WETH9_ {
...
         require(balanceOf[msg.sender] >= wad);
         balanceOf[msg.sender] -= wad;
         msg.sender.transfer(wad);
-        Withdrawal(msg.sender, wad);
+        emit Withdrawal(msg.sender, wad);
     }
 
     function totalSupply() public view returns (uint) {
-        return this.balance;
+      return address(this).balance;
```
Comment (asymmetric) [inline src/weth9.sol]: Got too little indent here.

**Comment by MrChico** [comment]
If no one opposed I will aim to get this merged tomorrow. (ping @rainbreak )

**Comment by asymmetric** [comment]
ping @levlivnev

**Pull request merged**